### PR TITLE
Service port for performance analyzer

### PR DESCRIPTION
**Pull request by Phoelsch**
Signed-off-by: Philipp Hölscher <phoelsch@outlook.de>

### Description
The change adds an additional port to the statefulset and service to expose the performance analyzer.
 
### Issues Resolved
[251 - [Enhancement][opensearch]add port 9600 (Performance Analyzer)](https://github.com/opensearch-project/helm-charts/issues/251)
 
### Check List
- [X ] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [X ] Helm chart version bumped
- [X ] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by TheAlgo** [comment]
@Phoelsch Thanks for this PR. Can you please check the contributing guidelines and refactor this PR? Thanks

**Comment by Phoelsch** [comment]
@TheAlgo I checked the guidlines even before creating the pull request. Can please give me a hint what`s missing or wrong?

**Comment by TheAlgo** [comment]
@Phoelsch Your commits are not signed

**Comment by Phoelsch** [comment]
@TheAlgo Thanks, commit is now signed.

File: charts/opensearch/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 ### Security
 ---
```
Comment (TheAlgo) [inline charts/opensearch/CHANGELOG.md:15]: Please add links to Unreleased section and release links at the bottom of CHANGELOG

File: charts/opensearch/values.yaml [hunk 2]
```diff
@@ -259,6 +259,7 @@ enableServiceLinks: true
 protocol: https
 httpPort: 9200
 transportPort: 9300
+metricsPort: 9600
```
Comment (TheAlgo) [inline charts/opensearch/values.yaml:262]: Can you also update values for one of the CI chart values?

**Review by TheAlgo — CHANGES_REQUESTED** [review]

File: charts/opensearch/values.yaml (hunk 2 above)
Comment (Phoelsch) [reply charts/opensearch/values.yaml:262]: Done.

File: charts/opensearch/CHANGELOG.md (hunk 1 above)
Comment (Phoelsch) [reply charts/opensearch/CHANGELOG.md:15]: Done.

**Review by TheAlgo — APPROVED** [review]
LGTM!

**Comment by peterzhuamazon** [comment]
I am good with this change @Phoelsch thanks for the contribution.
@prudhvigodithi please take a look on this.

Thanks.

**Comment by DandyDeveloper** [comment]
I actually wonder if we need to explicitly label this as `performanceAnalyzer` rather than metric.

Many people uniformally recognize metrics ports as prometheus compliant metrics. Aka, they can be pulled in via prometheus.

Additional, many people may be using the prometheus metric plugin to generate such alerts and have probably already added a port called metrics to achieve prometheus polling.

Could we change this to be more explicitly labeled as performance analyzer?

**Comment by peterzhuamazon** [comment]
We actually plan to remove the coupled dependency between OS and PA process:
* https://github.com/opensearch-project/opensearch-build/pull/2876
So think again, probably dont need to specifically specify this port?

**Comment by DandyDeveloper** [comment]
@peterzhuamazon Do we have the facilities in the chart to enable and include this for users by removing this?

Aka, can they independently define the plugins, ports etc. Necessary to at least not inadvertedly remove the functionality from our chart entirely?

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon Do we have the facilities in the chart to enable and include this for users by removing this?
> 
> Aka, can they independently define the plugins, ports etc. Necessary to at least not inadvertedly remove the functionality from our chart entirely?

Hi @DandyDeveloper 

The current change planned in that PR is that if user does not add anything then behavior is mostly same.
The PA and OS will both start, but this time killing PA process wont bring down OS process anymore.
If use define env var `DISABLE_PERFORMANCE_ANALYZER_AGENT_CLI=true` then PA will not start during the creation of the container.

User can modify the pa config to change port from 9600 to something else if needed, before starting container.
Thanks.

**Comment by TheAlgo** [comment]
> > One small question @Phoelsch if in a chart one does not provide the value for the port? Will it throw error? Will this break the chart?
> 
> The port is set to 9600 as default with values.yaml. If it`s not explicitly overwritten by the user it will still be 9600. I don`t see any problem with that.

@Phoelsch But if someone is upgrading with previous `values.yaml` , will it not break?

**Comment by Phoelsch** [comment]
> > > One small question @Phoelsch if in a chart one does not provide the value for the port? Will it throw error? Will this break the chart?
> > 
> > 
> > The port is set to 9600 as default with values.yaml. If it`s not explicitly overwritten by the user it will still be 9600. I don`t see any problem with that.
> 
> @Phoelsch But if someone is upgrading with previous `values.yaml` , will it not break?

@TheAlgo If I upgrade an existing deployment to this version the belonging  `values.yaml` with the new default value will be used.  How could it happen that the previous `values.yaml` will be used?

**Comment by TheAlgo** [comment]
> > > > One small question @Phoelsch if in a chart one does not provide the value for the port? Will it throw error? Will this break the chart?
> > > 
> > > 
> > > The port is set to 9600 as default with values.yaml. If it`s not explicitly overwritten by the user it will still be 9600. I don`t see any problem with that.
> > 
> > 
> > @Phoelsch But if someone is upgrading with previous `values.yaml` , will it not break?
> 
> @TheAlgo If I upgrade an existing deployment to this version the belonging `values.yaml` with the new default value will be used. How could it happen that the previous `values.yaml` will be used?

@Phoelsch Someone can still use the old `values.yaml` and issue an upgrade?

**Comment by TheAlgo** [comment]
> > > > > One small question @Phoelsch if in a chart one does not provide the value for the port? Will it throw error? Will this break the chart?
> > > > 
> > > > 
> > > > The port is set to 9600 as default with values.yaml. If it`s not explicitly overwritten by the user it will still be 9600. I don`t see any problem with that.
> > > 
> > > 
> > > @Phoelsch But if someone is upgrading with previous `values.yaml` , will it not break?
> > 
> > 
> > @TheAlgo If I upgrade an existing deployment to this version the belonging `values.yaml` with the new default value will be used. How could it happen that the previous `values.yaml` will be used?
> 
> @Phoelsch Someone can still use the old `values.yaml` and issue an upgrade?

@Phoelsch Opinions?

**Comment by TheAlgo** [comment]
@Phoelsch Can you check the comments wen free so that we can have a closure on this PR?

**Comment by prudhvigodithi** [comment]
Hey @Phoelsch this a nice PR to get merged, can you please fix the conflicts so that we can proceed to merge?
Thank you

**Comment by Phoelsch** [comment]
@TheAlgo I`m sorry for the delay. Still I don`t get you`re problem und could not reproduce that. We could use a default value within the statefulset template, but that would be awful. The Helm Docs itself says that the values.yaml should be used as I did.

**Comment by TheAlgo** [comment]
> @TheAlgo I'm sorry for the delay. Still I don't get you`re problem und could not reproduce that. We could use a default value within the statefulset template, but that would be awful. The Helm Docs itself says that the values.yaml should be used as I did.

Yes that might not be the right way. Lets have a minor version upgrade for this. What do you think? Tagging @prudhvigodithi  as well

**Review by TheAlgo — COMMENTED** [review]
Also lets resolve the conflicts as well! Thank you for this.

**Comment by TheAlgo** [comment]
@Phoelsch Any updates on this PR?

**Comment by Phoelsch** [comment]
> @Phoelsch Any updates on this PR?

Resolved

**Comment by Phoelsch** [comment]
> @Phoelsch Any updates on this PR?

@TheAlgo resolved

**Review by prudhvigodithi — APPROVED** [review]

**Review by bbarani — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Hey @Phoelsch this is a good change for 1.x version as well, can you please backport to 1.x branch?
Thank you

**Pull request merged**